From: Miklos Szeredi <miklos@szeredi.hu>
Subject: [FUSE] Fix compile warnings
Patch-mainline: 2.6.12-rc-mm

This patch fixes a couple of warnings when compiling on the x86_64
architecture.

Signed-off-by: Miklos Szeredi <miklos@szeredi.hu>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---

 fs/fuse/file.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff -puN fs/fuse/file.c~fuse-direct-i-o-fix-warning-on-x86_64 fs/fuse/file.c
--- 25/fs/fuse/file.c~fuse-direct-i-o-fix-warning-on-x86_64	2005-05-10 02:22:11.000000000 -0700
+++ 25-akpm/fs/fuse/file.c	2005-05-10 02:22:11.000000000 -0700
@@ -406,7 +406,7 @@ static ssize_t fuse_direct_io(struct fil
 {
 	struct inode *inode = file->f_dentry->d_inode;
 	struct fuse_conn *fc = get_fuse_conn(inode);
-	unsigned nmax = write ? fc->max_write : fc->max_read;
+	size_t nmax = write ? fc->max_write : fc->max_read;
 	loff_t pos = *ppos;
 	ssize_t res = 0;
 	struct fuse_req *req = fuse_get_request(fc);
@@ -414,8 +414,8 @@ static ssize_t fuse_direct_io(struct fil
 		return -ERESTARTSYS;
 
 	while (count) {
-		unsigned tmp;
-		unsigned nres;
+		size_t tmp;
+		size_t nres;
 		size_t nbytes = min(count, nmax);
 		int err = fuse_get_user_pages(req, buf, nbytes, !write);
 		if (err) {
_
